src/hooks/useForm.ts

```diff
@@ -51,18 +54,21 @@ export const useForm = ({ fields = [], onSubmit }: Dependencies) => {
    const [errors, setErrors] = useState<FormErr>({});
    const [isSubmit, setIsSubmit] = useState<boolean>(false);
 
-   // check if no errr , confirm to Submit
-   useEffect(() => {
-      if (isSubmit && Object.keys(errors).length === 0) {
-         onSubmit();
-      }
+   // // check if no error , confirm to Submit
+   // useEffect(() => {
+   //    if (isSubmit && Object.keys(errors).length === 0) {
+   //       onSubmit();
+   //    }
 
-      // eslint-disable-next-line
-   }, [errors, isSubmit, values]);
+   //    // eslint-disable-next-line
+   // }, [errors, isSubmit, values]);
 
    // handleSubmit
    const handleSubmit = (event) => {
       event.preventDefault();
+      if (isSubmit && Object.keys(errors).length === 0) {
+         onSubmit();
+      }
       setIsSubmit(true);
       setErrors(validateValue(values, fields));
    };
```

Comment:
cái này làm đơn giản thôi e, `isSubmit` ở đây đâu cần thiết đâu nhỉ, state này thừa nên e bị lỗi phải click 2 lần đấy
```suggestion
   const handleSubmit = (event) => {
      event.preventDefault();
      setErrors(validateValue(values, fields));
      if (Object.keys(errors).length === 0) {
         onSubmit();
      }
   };
```